### DISCUSSION TITLE
Static Web Apps:  Questions about Environments and Custom Domains

### DISCUSSION THREAD

**Discussion by RayDeanTech** [Azure/static-web-apps#1160] [Q&A]
**Q:**  Can Static Web Apps support new-school gTLDs  such as .tech, and .food?

**Q:**  What is considered a single use of a Custom domain?  Is the following example a use of 1 "custom domain"?  Or, is this 4 uses of the "custom domain"?

1. api.domain.tech
2. www.domain.tech
3. demo.domain.tech
4. dev.domain.tech

**Q:**  Can independent repositories be placed under specific hierarchies when deployed to the Static Web App?  By this, I mean:  assume all my content is at www.domain.com, but I want to have a separate repository for www.domain.com/demos/.  Would that be adjusted here?

/src == primary content (its own repository)
/src/demos == where demos go (a completely separate repository)
/src/YetAnotherRepository/ == another, completely different project

![image](https://user-images.githubusercontent.com/21373144/236630592-1a774601-d029-4559-aa82-32109d927ed3.png)

**Q:**  Do the included "3 pre-production environments" negate the need for a "dev" server?

![image](https://user-images.githubusercontent.com/21373144/236629978-982ef2cd-d7b0-456f-bd31-3823ec63c119.png)

**Comment by thomasgauvin** [comment]
Answers:

1. There should be no limitation regarding the TLDs, as long as you can properly configure them within your DNS provider
3. Those are 4 uses of the custom domain
4. You cannot deploy to a Static Web App 'path' independently. You would need to build your application properly with paths configured within your build/output, and deploy that build to SWA. In other words, a deployment to SWA cannot be partial, it must include the full contents. If you wanted to have independent repositories hosted under a single domain, we would recommend creating separate SWAs and handling routing with an AFD or gateway
5. You can use the pre-production environments as 'staging' or 'dev' environments indeed.
